### DISCUSSION TITLE
No MSG5 detected leads to `write_grid` failure later

### DISCUSSION THREAD

**Discussion by msteckle** [ARM-DOE/pyart#1396] [Q&A] [answered]
Hello! I've been using py-art to grab and grid some old AWS-stored NEXRAD Level II data. I did a run on post-2009 data and didn't run into any issues reading NEXRAD Level II data and writing it to netcdf after gridding. But I pushed my study period back to 2000 and started to get the `UserWarning: No MSG5 detected. Setting to meaningless data. Rethink your life choices and be ready for errors.Specifically fixed angle data will be missing`, which came from the `read_nexrad_archive` function. This issue did in fact make me rethink my life choices but also **had me wondering what MSG5 is and why it was messing up my process to export gridded radar as netcdfs**. 

At first, I didn't know why my grids were failing to convert to netcdfs properly, but I looked back at the failed ones and during the `read_nexrad_archive` process I got that warning, which I didn't get on my successful grid-to-netcdfs. Sorry, I don't have the full error message when using `write_grid` saved to show you.

Is there a way to fix the MSG5 issue? I would assume not an easy one, or that would have been built into the `read_nexrad_archive` function already. **Either way, what is the MSG5 issue, and why did it happen so much with old radar data?** I couldn't find any explanation on the internet. Thanks!

**Comment by jrwebster** [comment]
Can you identify examples of the AWS hosted NEXRAD Level II data that result in the MSG5 issue?

**Reply by msteckle** [reply]
Sure, here is one that gave me that warning: `2001/11/27/KOHX/KOHX20011127_052231.gz`

**Comment by dopplershift** [comment, marked as answer]
Message 5 in NEXRAD Level 2 format contains the VCP (Volume Coverage Pattern) information. I believe PyART relies on this to know what scanning parameters to expect, especially when it comes to the elevation angles for sweeps.

The funny error message aside, it's really expected for the older data. I don't have documentation on when, but that message started being included it looks like somewhere in the ~2006 timeframe. NEXRAD Level2 format was originally distributed around on tapes and not intended as a file format, and it wasn't readily available on the internet until the early 2000s. It took awhile for all the useful information types (messages) to be included.

**Reply by msteckle** [reply]
Excellent info, thank you!

**Comment by zssherman** [comment]
Hello! To add to dopplershift message, I believe Py-ART uses this information and fixed angle for gate coord information etc. (was digging through the code, been awhile haha) but when gridding, it uses the coordinate information to interpolate to a Cartesian grid and my guess is the error is zero coordinate info for the gates causing some issues. Sadly i'm not sure a way around it. I know for datasets outside of nexrad, (users with certain data formats not readable with Py-ART) they usually would create an ingest, add fixed angle to the radar object that recalculate the gate coordinate info with radar.init_gate_latitude_longitude(), but not sure if this is entirely possible in this case sadly. Someone else might have more to add

**Reply by msteckle** [reply]
Thanks for looking to see why that MSG5 warning led to gridding failure later on! For now, I've just dropped scans with that warning, and it didn't impact the breadth of my dataset horrifically. So, I'm not too sad to not have a fix for it. Thanks again!
